src/DynamoPublish/Models/PublishModel.cs

```diff
@@ -0,0 +1,95 @@
+﻿using Dynamo.Core;
+using Dynamo.Interfaces;
+using Dynamo.Models;
+using Dynamo.Nodes;
+using Dynamo.Wpf.Authentication;
+using Greg;
+using Greg.AuthProviders;
+using Reach;
+using System;
+using System.Collections.Generic;
+using System.Configuration;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+
+namespace Dynamo.Publish.Models
+{
+    public class PublishModel
+    {
+        private readonly IAuthProvider authenticationProvider;
+
+        private readonly string serverUrl;
+        private readonly string port;
+        private readonly string page;
+
+        public bool IsLoggedIn
+        {
+            get
+            {
+                return authenticationProvider.LoginState == LoginState.LoggedIn;
+            }
+        }
+
+        #region Initialization
+
+        public PublishModel(IAuthProvider dynamoAuthenticationProvider)
+        {
+            // Open the configuration file using the dll location.
+            Configuration config = ConfigurationManager.OpenExeConfiguration(this.GetType().Assembly.Location);
+            // Get the appSettings section.
+            AppSettingsSection appSettings = (AppSettingsSection)config.GetSection("appSettings");
+
```

Comment:
Just a general suggestion, prefer `var` to explicitly declaring the type.  It makes it easier to refactor...